guides/source/action_controller_overview.md

```diff
@@ -396,145 +436,163 @@ end
+NOTE: Be aware that JSON has limited support serializing Ruby objects suck as
+`Date`, `Time`, and `Symbol`. These will be serialized and deserialized into
+`String`s:
 
 ```ruby
-# Be sure to restart your server when you modify this file.
-Rails.application.config.session_store :cookie_store, key: '_your_app_session'
-```
-
-You can also pass a `:domain` key and specify the domain name for the cookie:
+class CookiesController < ApplicationController
+  def set_cookie
+    cookies.encrypted[:expiration_date] = Date.tomorrow # => Thu, 20 Mar 2014
+    redirect_to action: 'read_cookie'
+  end
 
-```ruby
-# Be sure to restart your server when you modify this file.
-Rails.application.config.session_store :cookie_store, key: '_your_app_session', domain: ".example.com"
+  def read_cookie
+    cookies.encrypted[:expiration_date] # => "2014-03-20"
+  end
+end
 ```
 
-Rails sets up (for the CookieStore) a secret key used for signing the session data in `config/credentials.yml.enc`. This can be changed with `bin/rails credentials:edit`.
-
-```yaml
-# aws:
-#   access_key_id: 123
-#   secret_access_key: 345
-
-# Used as the base secret for all MessageVerifiers in Rails, including the one protecting cookies.
-secret_key_base: 492f...
-```
+If you need to store these or more complex objects, you may need to manually
+convert their values when reading them in subsequent requests.
 
-NOTE: Changing the secret_key_base when using the `CookieStore` will invalidate all existing sessions.
+If you use the cookie session store, the above applies to the `session` and
+`flash` hash as well.
 
-[`ActionDispatch::Session::CookieStore`]: https://api.rubyonrails.org/classes/ActionDispatch/Session/CookieStore.html
-[`ActionDispatch::Session::CacheStore`]: https://api.rubyonrails.org/classes/ActionDispatch/Session/CacheStore.html
-[`ActionDispatch::Session::MemCacheStore`]: https://api.rubyonrails.org/classes/ActionDispatch/Session/MemCacheStore.html
-[activerecord-session_store]: https://github.com/rails/activerecord-session_store
+[`config.action_dispatch.cookies_serializer`]: configuring.html#config-action-dispatch-cookies-serializer
+[`cookies`]: https://api.rubyonrails.org/classes/ActionController/Cookies.html#method-i-cookies
 
+Session
+-------
 
-### Accessing the Session
+While cookies are stored client-side, session data is stored server-side (in
+memory, a database, or a cache), and the duration is usually temporary and tied
+to the user's session (e.g. until they close the browser). An example use case
+for session is storing sensitive data like user authentication.
 
-In your controller, you can access the session through the `session` instance method.
+In a Rails application, the session is available in the controller and the view.
 
-NOTE: Sessions are lazily loaded. If you don't access sessions in your action's code, they will not be loaded. Hence, you will never need to disable sessions, just not accessing them will do the job.
+### Working With the Session
```

Comment:
Since "with" is a [preposition](https://edgeguides.rubyonrails.org/ruby_on_rails_guides_guidelines.html#headings) here, should it be 

```suggestion
### Working with the Session
```